### PR TITLE
Base homebrew/brew Docker image on ubuntu:20.04

### PR DESCRIPTION
**Pull request by sjackman**
Add a new image `homebrew/ubuntu16.04` for building Linux bottles.
Tag the most recent stable release of each image as `latest`.

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

`homebrew/ubuntu16.04` has two symbolic tags. `latest` is the most recent stable release of `Homebrew/brew`. `master` is the `master` branch of `Homebrew/brew`.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -33,9 +33,20 @@ jobs:
...
           docker push "homebrew/ubuntu${{matrix.version}}:$brew_version"
+          docker tag brew "homebrew/ubuntu${{matrix.version}}:latest"
+          docker push "homebrew/ubuntu${{matrix.version}}:latest"
+      - name: Deploy the homebrew/brew Docker image to GitHub and Docker Hub
+        if: startsWith(github.ref, 'refs/tags/') && matrix.version == '20.04'
+        run: |
+            docker tag brew "docker.pkg.github.com/homebrew/brew/brew:latest"
+            docker push "docker.pkg.github.com/homebrew/brew/brew:latest"
+            docker tag brew "docker.pkg.github.com/homebrew/brew/brew:latest"
+            docker push "docker.pkg.github.com/homebrew/brew/brew:latest"
```
Comment (Bo98) [inline .github/workflows/docker.yml:52]: Is one ment to be `$brew_version` or Docker Hub?

Comment (sjackman) [reply .github/workflows/docker.yml]: 🙈 Thanks for catching this error, Bo!

**Comment by tseemann** [comment]
16.04 was April 2016 , so 5 year support is April 2021 right?

**Comment by sjackman** [comment]
@MikeMcQuaid wrote…

> What was the old version and what are the consequences of this?

Homebrew/brew was based on Ubuntu 16.04. The new image Homebrew/ubuntu16.04 is based on Ubuntu 16.04. Effectively Homebrew/brew has been renamed to Homebrew/ubuntu16.04. We will continue to use Homebrew/ubuntu16.04 for bottling.

There is no consequence for Homebrew maintainers, except that we’ll change `image: homebrew/brew` to `image: homebrew/ubuntu16.04` in our GitHub Actions configuration.

There are two advantages for users who are using `homebrew/brew`. This image will now be be based on Ubuntu 20.04 rather than Ubuntu 16.04 The `latest` tag will now be the most recent _stable_ release of Homebrew, not the bleeding edge `master`.

`homebrew/ubuntu16.04` has two symbolic tags. `latest` is the most recent stable release. `master` is, well, the `master` branch.

**Comment by MikeMcQuaid** [comment]
@sjackman All sounds great/sensible to me.

**Pull request merged**